pkg/healthcheck/healthcheck.go

```diff
@@ -1463,6 +1546,169 @@ func FetchLinkerdConfigMap(k kubernetes.Interface, controlPlaneNamespace string)
 	return cm, configPB, nil
 }
 
+func (hc *HealthChecker) checkServiceMirrorController() error {
+	options := metav1.ListOptions{
+		LabelSelector: hc.serviceMirrorComponentsSelector(),
+	}
+	result, err := hc.kubeAPI.AppsV1().Deployments(corev1.NamespaceAll).List(options)
+	if err != nil {
+		return err
+	}
+
+	if len(result.Items) < 1 {
+		return errors.New("Service mirror controller is not present")
+	}
+
+	if len(result.Items) > 1 {
+		var errors []string
+		for _, smc := range result.Items {
+			errors = append(errors, fmt.Sprintf("%s/%s", smc.Namespace, smc.Name))
+		}
+		return fmt.Errorf("There are more than one service mirror controllers:\n\t%s", strings.Join(errors, "\n\t"))
+
+	}
+
+	controller := result.Items[0]
+	if controller.Status.AvailableReplicas < 1 {
+		return fmt.Errorf("Service mirror controller is not available: %s/%s", controller.Namespace, controller.Name)
+	}
+	hc.serviceMirrorNs = controller.Namespace
+	return nil
+}
+
+func (hc *HealthChecker) checkServiceMirrorLocalRBAC() error {
+	compare := func(expected, actual []string) error {
+		sort.Strings(expected)
+		sort.Strings(actual)
+
+		expectedStr := strings.Join(expected, ",")
+		actualStr := strings.Join(actual, ",")
+
+		if expectedStr != actualStr {
+			return fmt.Errorf("expected %s, got %s", expectedStr, actualStr)
+		}
+
+		return nil
+	}
+
+	cr, err := hc.kubeAPI.RbacV1().ClusterRoles().Get(linkerdServiceMirrorComponentName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Could not obtain service mirror ClusterRole: %s", err)
+	}
+
+	var policyRule *v1.PolicyRule
+
+	for _, r := range cr.Rules {
+		if len(r.APIGroups) == 1 && r.APIGroups[0] == "" {
+			rule := r
+			policyRule = &rule
+		}
+	}
+
+	if policyRule == nil {
+		return fmt.Errorf("Service mirror ClusterRole is missing expected policy rule")
+	}
+
+	if err := compare(expectedServiceMirrorPolicyVerbs, policyRule.Verbs); err != nil {
+		return fmt.Errorf("Service mirror ClusterRole is missing verbs: %s", err)
+	}
+
+	if err := compare(expectedServiceMirrorPolicyResources, policyRule.Resources); err != nil {
+		return fmt.Errorf("Service mirror ClusterRole is missing required resources: %s", err)
+	}
```

Comment:
Can you avoid short-circuiting on the first error here, and instead show both errors if present?